File: CafeEase/CafeEase.Services/Database/Notification.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CafeEase.Services.Database
{
    public class Notification
    {
        public int Id { get; set; }

        public int UserId { get; set; }
        public virtual User User { get; set; } = null!;

        public int? OrderId { get; set; }

        public string Title { get; set; } = null!;
        public string Body { get; set; } = null!;

        public bool IsRead { get; set; } = false;
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: CafeEase/UI/cafeease_desktop/lib/models/notification.dart
import 'package:json_annotation/json_annotation.dart';
part 'notification.g.dart';

@JsonSerializable()
class AppNotification {
  int? id;
  int? userId;
  int? orderId;
  String? title;
  String? body;
  bool? isRead;
  DateTime? createdAt;

  AppNotification({this.id, this.userId, this.orderId, this.title, this.body, this.isRead, this.createdAt});

  factory AppNotification.fromJson(Map<String, dynamic> json) => _$AppNotificationFromJson(json);
  Map<String, dynamic> toJson() => _$AppNotificationToJson(this);
}